src/Link_TEST.cc

```diff
@@ -763,12 +766,121 @@ TEST(DOMLink, InertialValuesGivenWithAutoSetToTrue)
   root.ResolveAutoInertials(errors, sdfParserConfig);
   EXPECT_TRUE(errors.empty());
 
-  EXPECT_NE(4.0, link->Inertial().MassMatrix().Mass());
+  EXPECT_EQ(2.0, link->Inertial().MassMatrix().Mass());
   EXPECT_EQ(gz::math::Pose3d::Zero, link->Inertial().Pose());
   EXPECT_EQ(gz::math::Vector3d(0.33333, 0.33333, 0.33333),
     link->Inertial().MassMatrix().DiagonalMoments());
 }
 
+/////////////////////////////////////////////////
+TEST(DOMLink, ResolveAutoInertialsWithMass)
+{
+  // A model with link inertial auto set to true.
+  // The inertia matrix is specified but should be ignored.
+  // <mass> is speicifed - the auto computed inertial values should
+  // be scaled based on the desired mass.
+  std::string sdf = "<?xml version=\"1.0\"?>"
+  "<sdf version=\"1.11\">"
+  "  <model name='compound_model'>"
+  "   <link name='compound_link'>"
+  "     <inertial auto='true'>"
+  "       <mass>4.0</mass>"
+  "       <pose>1 1 1 2 2 2</pose>"
+  "       <inertia>"
+  "         <ixx>1</ixx>"
+  "         <iyy>1</iyy>"
+  "         <izz>1</izz>"
+  "       </inertia>"
+  "     </inertial>"
+  "     <collision name='box_collision'>"
+  "      <density>2.0</density>"
```

Comment:
@azeey I think we could delete this line in order to test without density being set

Reply:
added new test without density being set in https://github.com/gazebosim/sdformat/pull/1517